pkg/apis/ndm/v1alpha1/types.go

```diff
@@ -71,3 +71,37 @@ type StoragePoolList struct {
 
 	Items []StoragePool `json:"items"`
 }
+
+// +genclient
+// +genclient:noStatus
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +resource:path=diskobject
+
+// DiskObject describes disk resource.
+type DiskObject struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata, omitempty"`
+
+	Spec DiskObjectSpec `json:"spec"`
+}
+
+// DiskObjectSpec is the specification for the disk stored as CRD
+type DiskObjectSpec struct {
+	Uuid       string `json:"uuid"`  //unique identifier for the disk
```

Comment:
How is this uuid determined? We should use GUID of the disk, they are designed for that and it allows us to determine that disks are moved (hot swapped) between nodes as well.  I'm referring to the NAA IEEE number see SPC4 section 7.

Reply:
we can discuss this on the design. Please refer https://github.com/openebs/node-disk-manager/pull/1